negotiation/src/main/java/it/eng/negotiation/exception/ContractNegotiationExceptionAdvice.java

```diff
@@ -25,21 +30,37 @@ protected ResponseEntity<Object> handleContractNotFound(ContractNegotiationNotFo
                 .code(HttpStatus.NOT_FOUND.getReasonPhrase())
                 .reason(Collections.singletonList(Reason.Builder.newInstance().language("en").value(ex.getLocalizedMessage()).build()))
                 .description(Collections.singletonList(Description.Builder.newInstance().language("en").value(ex.getLocalizedMessage()).build())).build();
-        String response = Serializer.serializeProtocol(errorMessage);
 
-        return handleExceptionInternal(ex, response, new HttpHeaders(), HttpStatus.NOT_FOUND, request);
+        return handleExceptionInternal(ex, Serializer.serializeProtocolJsonNode(errorMessage), new HttpHeaders(), HttpStatus.NOT_FOUND, request);
     }
+    
+    @ExceptionHandler(value = {ValidationException.class})
+    protected ResponseEntity<Object> handleValidationException(ValidationException ex, WebRequest request) {
+    	  ContractNegotiationErrorMessage errorMessage = ContractNegotiationErrorMessage.Builder.newInstance()
+    			//TODO add proper provider and consumer pid
+    			  .providerPid(createNewId())
+    			  .consumerPid(createNewId())
+                  .code(HttpStatus.BAD_REQUEST.getReasonPhrase())
+                  .reason(Collections.singletonList(Reason.Builder.newInstance().language("en").value(ex.getLocalizedMessage()).build()))
+                  .description(Collections.singletonList(Description.Builder.newInstance().language("en").value(ex.getLocalizedMessage()).build())).build();
+    	  
+    	  return handleExceptionInternal(ex, Serializer.serializeProtocolJsonNode(errorMessage), new HttpHeaders(), HttpStatus.BAD_REQUEST, request);
+    }
+    
 
     @ExceptionHandler(value = {ContractNegotiationExistsException.class})
     protected ResponseEntity<Object> handleContractExists(ContractNegotiationExistsException ex, WebRequest request) {
 
         ContractNegotiationErrorMessage errorMessage = ContractNegotiationErrorMessage.Builder.newInstance()
                 .providerPid(ex.getProviderPid())
-                .code(HttpStatus.NOT_FOUND.getReasonPhrase())
+                .code(HttpStatus.CONFLICT.getReasonPhrase())
                 .reason(Collections.singletonList(Reason.Builder.newInstance().language("en").value(ex.getLocalizedMessage()).build()))
                 .description(Collections.singletonList(Description.Builder.newInstance().language("en").value(ex.getLocalizedMessage()).build())).build();
-        String response = Serializer.serializeProtocol(errorMessage);
 
-        return handleExceptionInternal(ex, response, new HttpHeaders(), HttpStatus.NOT_FOUND, request);
+        return handleExceptionInternal(ex, Serializer.serializeProtocolJsonNode(errorMessage), new HttpHeaders(), HttpStatus.CONFLICT, request);
```

Comment:
Is this conflict status according to DSP specification?

Reply:
You were right, I changed it